spec/requests/api/providers_spec.rb

```diff
@@ -130,6 +146,24 @@
       expect(response).to have_http_status(:forbidden)
     end
 
+    it "creates provider with custom attributes" do
+      api_basic_authorize collection_action_identifier(:providers, :create)
+      run_post(providers_url, sample_rhevm.merge!("custom_attributes" => custom_attributes))
```

Comment:
@alongoldboim I think:
1) is possible - you ojust pass in the resources array
2) You are free to override the default implementation in the controller for those actions
3) As with 2